### PR TITLE
[sqlite3] update to <3.37.1>

### LINKED ISSUE #22353: [sqlite3] update to <3.37.1>
Library name: sqlite3

New version number: 3.37.1

Other information that may be useful (release notes, etc...)
https://sqlite.org/releaselog/3_37_1.html

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #22353 
[sqlite3] update to 3.37.1
All features have been tested successfullly in the following triplet:

- x64-windows
- x64-windows-static
- x86-windows

### CODE REVIEW

File: ports/ogre-next/portfile.cmake [hunk 1]
```diff
@@ -29,10 +29,11 @@ else()
...
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-    d3d9    OGRE_BUILD_RENDERSYSTEM_D3D9
-    java    OGRE_BUILD_COMPONENT_JAVA
-    python  OGRE_BUILD_COMPONENT_PYTHON
-    csharp  OGRE_BUILD_COMPONENT_CSHARP
+    FEATURES
+        d3d9    OGRE_BUILD_RENDERSYSTEM_D3D9
+        java    OGRE_BUILD_COMPONENT_JAVA
+        python  OGRE_BUILD_COMPONENT_PYTHON
+        csharp  OGRE_BUILD_COMPONENT_CSHARP
```
Comment (FrankXie05) [inline ports/ogre-next/portfile.cmake:36]: Depend on [ogre-next] FEATURES

File: ports/ogre-next/portfile.cmake [hunk 2]
```diff
@@ -29,10 +29,11 @@ else()
...
 )
 
 vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/ogre-next/portfile.cmake:39]: Could you modernize this cmake function？

File: ports/ogre-next/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ogre-next",
   "version-string": "2019-10-20",
```
Comment (JonLiu1993) [inline ports/ogre-next/vcpkg.json:3]:
```suggestion
  "version-date": "2019-10-20",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/ogre-next/portfile.cmake [hunk 4]
```diff
@@ -29,15 +29,15 @@ else()
...
+    FEATURES
+        d3d9    OGRE_BUILD_RENDERSYSTEM_D3D9
+        java    OGRE_BUILD_COMPONENT_JAVA
+        python  OGRE_BUILD_COMPONENT_PYTHON
+        csharp  OGRE_BUILD_COMPONENT_CSHARP
 )
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/ogre-next/portfile.cmake:40]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

**Comment by dg0yt** [comment]
IMO the ogre changes should be a separate PR.

**Comment by FrankXie05** [comment]
@dg0yt #22058  I have submitted the pr of ogre-next update, and it is waiting for review. And I'll just merge it by then  :)

**Comment by dg0yt** [comment]
> @dg0yt #22058 I have submitted the pr of ogre-next update, and it is waiting for review. And I'll just merge it by then :)

I still miss the link why there are ogre-next changes in this sqlite3 PR.

**Comment by FrankXie05** [comment]
@dg0yt yes it is my problem, it really should be modified in ogre-next instead of sqlite3.  Thanks. :)

**Comment by FrankXie05** [comment]
Fix by #22795

**Pull request closed without merging**